test/functional/test-amp-video.js

```diff
@@ -16,27 +16,46 @@
 
 import {createIframePromise} from '../../testing/iframe';
 import {installVideo} from '../../builtins/amp-video';
+import {installViewerService} from '../../src/service/viewer-impl';
 
 describe('amp-video', () => {
 
+  let sandbox;
+
+  beforeEach(() => {
+    sandbox = sinon.sandbox.create();
+  });
+
+  afterEach(() => {
+    sandbox.restore();
+    sandbox = null;
+  });
+
   function getFooVideoSrc(mediatype) {
     return '//someHost/foo.' + mediatype.slice(mediatype.indexOf('/') + 1); // assumes no optional params
   }
 
-  function getVideo(attributes, children) {
-    return createIframePromise().then(iframe => {
-      installVideo(iframe.win);
-      const v = iframe.doc.createElement('amp-video');
-      for (const key in attributes) {
-        v.setAttribute(key, attributes[key]);
-      }
-      if (children != null) {
-        for (const key in children) {
-          v.appendChild(children[key]);
-        }
-      }
-      return iframe.addElement(v);
-    });
+  function getVideo(attributes, children, opt_visibilityState,
+                    opt_beforeLayoutCallback) {
+    return createIframePromise(
+        undefined, opt_beforeLayoutCallback).then(iframe => {
+          if (opt_visibilityState) {
+            const viewer = installViewerService(iframe.win);
+            sandbox.stub(viewer, 'isVisible').returns(
```

Comment:
Nope. Dropped it. Was trying different ways of testing `buildCallback` before `layoutCallback` is being called.